### DISCUSSION TITLE
Lame Duck Mode

### DISCUSSION THREAD

**Discussion by zakk616** [nats-io/nats-server#6070] [Q&A]
I am testing restart nats-server during load using ldm with command `nats-server --signal ldm` and checking the nats-server.service status using `systemctl status nats-server.service` which says **Active: failed** instead of **Active: incative**

![image](https://github.com/user-attachments/assets/92fbc007-5465-4237-a4eb-ded887ed4bf6)

**NATS Version:** 2.10.19
**Note:** I am using the default `lame_duck_duration: 2m` and `lame_duck_grace_period: 10s`

**Comment by derekcollison** [comment]
If the server does not have alot of client connections it can exit earlier then lame duck duration. If it has no clients it could exit immediately.

**Reply by derekcollison** [reply]
Would need more info on the server's state before you issue the signal.

**Reply by zakk616** [reply]
server has only 14 connections, still its not being stopped gracefully and terminating.
neither `nats-server --signal ldm` works, nor `systemctl stop nats-server.service` works when ldm is defined in config file.

though when I change the service line:
`ExecStop=/bin/kill -s SIGINT $MAINPID`
to line:
`ExecStop=/bin/kill -V -s SIGINT $MAINPID`

Only it works fine, strange!

I am not sure its the OS level issue or at nats-server level.

**Reply by ripienaar** [reply]
I think if you're setting a `ExecStop` in the unit file you probably shouldn't make it exit using `nats-server --signal` but always using the `systemctl` command.
